Baikal/Kernels/CL/mipmap_level_scaler.cl

```diff
@@ -0,0 +1,261 @@
+/**********************************************************************
+Copyright (c) 2018 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#ifndef MIPMAP_LEVEL_SCALER_CL
+#define MIPMAP_LEVEL_SCALER_CL
+
+#include <../Baikal/Kernels/CL/common.cl>
+#include <../Baikal/Kernels/CL/payload.cl>
+#include <../Baikal/Kernels/CL/utils.cl>
+#include <../Baikal/Kernels/CL/texture.cl>
+
+#pragma OPENCL EXTENSION cl_khr_fp16 : enable
+
+KERNEL
+void ComputeWeights_NoRounding(
+    GLOBAL float3* restrict weights,
+    // Size of weight buffer
+    const int size)
+{
+    int id = get_global_id(0);
+
+    if (id < size)
+    {
+        weights[id].x = .0f;
```

Comment:
why you need this kernel?

Reply:
also weights[id] = (float3)(0.f,  0.5f, 0.5f);
Why 0.f for x and 0.5f for y?

Reply:
The kernel is used when size (width/height) of current mip level can be divided by two without remainder (is even).

In general the convolution formula is:
> next_level[x] = w0 * curr_level[2 * x - 1] +  w1 * curr_level[2 * x] + w2 * curr_level[2 * x + 1]

When the texture size is odd, we apply simpler version:
> next_level[x] = 0.5 * curr_level[2 * x] + 0.5 *curr_level[2 * x + 1]

But if the size of mip level is odd, we divide it by two and round up. In this case the convolution weights are:
> w0 = (size - x - 1) / (2 * n - 1)
w1 = size / (2 * n - 1)
w2 = x / (2 * n - 1)

As we can see, in the first case the x component is not being used, so we set it to zero.